pkg/lib/cockpit-components-modifications.jsx

```diff
@@ -141,6 +141,8 @@ export class Modifications extends React.Component {
 
     render() {
         let emptyRow = null;
+        let fail_message = this.props.permitted ? _("No System Modifications") : _("The logged in user is not permitted to view system modifications");
+        fail_message = this.props.failed ? _("Loading of system modifications failed") : fail_message;
```

Comment:
Nitpick: This feels upside down -- as this.props.failed has the highest priority, it should be tested first.

Reply:
Yes, it may seem wierd, but it is two lines and has logic that it sets up default message and if failed it overwrites it as it has priority.

Other case would be:
```
let fail_message = _("Loading of system modifications failed");
if (!this.props.failed)
     fail_message = this.props.permitted ? _("No System Modifications") : _("The logged in user is not permitted to view system modifications");
```
Which I think is a bit stranger, because you set up fail message and immediately check if it did not fail.

The proper way would be:
```
let failed_message = ""
if (this.props.failed)
    failed_message = _("Loading of system modifications failed");
else
    fail_message = this.props.permitted ? _("No System Modifications") : _("The logged in user is not permitted to view system modifications");
```
(or even better not to have the ternary operator there but move it into the condition through `if/elif/else`)
It is pointlessly long for such simple condition. But surely I can go with it if you think it is better.

Reply:
Right, I was thinking of the latter, but let's not dwell on that. It's clear enough. Thanks!